### ISSUE TITLE
Use "publishdate" as default for "date" ?

### ISSUE THREAD

**Issue by jcw** [gohugoio/hugo#2145]
A small feature request, if I may...

I write all my posts in advance, and then publish them on specific days at midnight (I've been doing this for years, switched from WP to Hugo a few months ago, and am so glad I finally did! - FWIW). I'm about to streamline the process by having Hugo regenerate and upload the weblog daily at midnight.

This can be automated by entering a `publishdate = "..."`, which is great. But I still need to enter the `date = "..."` field as well.

Would it be possible to auto-set the `date` field from the `publishdate`, if none is present?
Or to put it differently: make the `date` field default to `publishdate`?
Right now, omitting the `date` value will timestamp the posts as Jan, 0001.

Assuming I'm not overlooking some feature in Hugo.
It's a small change, I hope - but just that little nudge extra to make posting more enjoyable still...

Cheers,
-jcw

PS. And a minor minor bug report, I think: the "9 of 0" below should perhaps be "9 of 9"?

```
Change detected, rebuilding site
2016-05-13 21:04 +0200
0 draft content
9 of 0 future rendered
132 pages created
[...]
```

**Comment by bep** [comment]
Yes that makes sense.

**Comment by g3wanghc** [comment]
`msg = fmt.Sprintf("%d future rendered", s.draftCount)` should be `futureCount` in `site.go`.

I will fix it. :V

**Comment by g3wanghc** [comment]
@bep Future rendered bug fix on 836e3a7

**Comment by mubix** [comment]
Agreed, I'd love not to have the exact same data in my posts twice (more chance for me to mess them up)

**Comment by kaushalmodi** [comment]
+1 for this feature.

In addition to what the OP said, if both data and publishdate are set in the frontmatter, then please set `.Date` to return the publishdate.

**Comment by kaushalmodi** [comment]
I have found this workaround for now: https://discuss.gohugo.io/t/how-to-use-the-publishdate-if-both-publishdate-and-date-are-set-in-frontmatter/5142/2?u=kaushalmodi

**Comment by graemebenzie** [comment]
Line 31 https://github.com/netlify/netlify-cms/blob/caa5c69522e5446e8677775d7ea25fdbb0e2b36e/src/backends/backend.js

I found that I got this error because I had removed the Title field for the collection.

**Comment by vassudanagunta** [comment]
This is the current code:

```golang
	if p.Date.IsZero() {
		p.Date = p.PublishDate
	}

	if p.PublishDate.IsZero() {
		p.PublishDate = p.Date
	}
```

So shouldn't this be marked as fixed?

**Comment by kaushalmodi** [comment]
@bep @vassudanagunta 

I think the issue is that the `date` in `.Page.Params` is left unset if only `publishdate` is set.

### Test case

- [Markdown][1] | [Hugo output][2]

Notice the **Page Params (Debug)**  section in the Hugo output linked above. Just as `lastmod` is auto-set based on `publishdate`, the request is that `date` also be auto-set if `publishdate` is specified, but `date` is not specified explicitly in the front-matter. 

[1]: https://gitlab.com/kaushalmodi/hugo-sandbox/raw/6cfeb93e9db58059cd9713c8237b15b95beeee93/content/bugs/date-and-publishdate.md
[2]: https://hugo-sandbox.netlify.com/bugs/date-and-publishdate/#page-debug

**Comment by vassudanagunta** [comment]
@kaushalmodi 

Since this issue is nearly 2 years old, and was fixed by #3854 per its title and description quite a while ago, I'd say create a new issue for what you describe.

**Comment by kaushalmodi** [comment]
@vassudanagunta I can create a new issue, but the test case I linked simply shows that the request in the original post of this issue is not met:

> Would it be possible to auto-set the `date` field from the `publishdate`, if none is present?
